From 848c9ffaca9a529e3eb077cbc30db9fe47517484 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Thu, 16 Oct 2014 16:01:44 -0400
Subject: [scsi] mpt2sas: Don't disable device twice at suspend

Message-id: <1413475335-14737-2-git-send-email-thenzl@redhat.com>
Patchwork-id: 97584
O-Subject: [RHEL7.1 e-stor PATCH 01/32] mpt2sas: Don't disable device twice at suspend.
Bugzilla: 1088524
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

On suspend, _scsih_suspend calls mpt2sas_base_free_resources, which
in turn calls pci_disable_device if the device is enabled prior to
suspending. However, _scsih_suspend also calls pci_disable_device
itself.

Thus, in the event that the device is enabled prior to suspending,
pci_disable_device will be called twice. This patch removes the
duplicate call to pci_disable_device in _scsi_suspend as it is both
unnecessary and results in a kernel oops.

Signed-off-by: Tyler Stachecki <tstache1@binghamton.edu>
Cc: stable@vger.kernel.org
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit af61e27c3f77c7623b5335590ae24b6a5c323e22)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/mpt2sas/mpt2sas_scsih.c b/drivers/scsi/mpt2sas/mpt2sas_scsih.c
index 7351843..5055f92 100644
--- a/drivers/scsi/mpt2sas/mpt2sas_scsih.c
+++ b/drivers/scsi/mpt2sas/mpt2sas_scsih.c
@@ -8287,7 +8287,6 @@ _scsih_suspend(struct pci_dev *pdev, pm_message_t state)
 
  mpt2sas_base_free_resources(ioc);
  pci_save_state(pdev);
- pci_disable_device(pdev);
  pci_set_power_state(pdev, device_state);
  return 0;
 }
-- 
1.7.1